### PR TITLE
support env file

### PR DESCRIPTION
**Pull request by ttionya**

### CODE REVIEW

File: scripts/entrypoint.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ if [[ "$1" == "mail" ]]; then
     MAIL_SMTP_ENABLE="TRUE"
     MAIL_DEBUG="TRUE"
 
-    if [[ -n "$2" ]]; then
```
Comment (mustaphazorgati) [inline scripts/entrypoint.sh:17]: I don't like this change. Even though they do the same I prefer the `-n` for readability :)

File: scripts/includes.sh [hunk 2]
```diff
@@ -83,6 +84,55 @@ function send_mail_content() {
...
 }
 
+########################################
+# Export variables from .env file.
+# Arguments:
+#     None
+# Outputs:
+#     variables with prefix 'DOTENV_'
+# Reference:
+#     https://gist.github.com/judy2k/7656bfe3b322d669ef75364a46327836
```
Comment (mustaphazorgati) [inline scripts/includes.sh:94]: Please replace with this: https://gist.github.com/judy2k/7656bfe3b322d669ef75364a46327836#gistcomment-3625311

I had to search that comment which is uncomforatble. This link jumps directly to that comment

File: scripts/includes.sh [hunk 3]
```diff
@@ -83,6 +84,55 @@ function send_mail_content() {
...
+
+    if [[ "${!VAR:-}" ]]; then
+        VALUE="${!VAR}"
+    elif [[ "${!VAR_FILE:-}" ]]; then
+        VALUE="$(cat "${!VAR_FILE}")"
+    elif [[ "${!VAR_DOTENV_FILE:-}" ]]; then
+        VALUE="$(cat "${!VAR_DOTENV_FILE}")"
+    elif [[ "${!VAR_DOTENV:-}" ]]; then
+        VALUE="${!VAR_DOTENV}"
+    fi
```
Comment (mustaphazorgati) [inline scripts/includes.sh:131]: Please use `-n` for readability

File: scripts/includes.sh [hunk 4]
```diff
@@ -83,6 +84,55 @@ function send_mail_content() {
...
+#     variable value
+########################################
+function get_env() {
+    local VAR="$1"
+    local VAR_FILE="${VAR}_FILE"
+    local VAR_DOTENV="DOTENV_${VAR}"
+    local VAR_DOTENV_FILE="DOTENV_${VAR_FILE}"
+    local VALUE=""
+
+    if [[ "${!VAR:-}" ]]; then
```
Comment (mustaphazorgati) [inline scripts/includes.sh:123]: what does the `!` do?

File: scripts/entrypoint.sh (hunk 1 above)
Comment (ttionya) [reply scripts/entrypoint.sh:17]: Yes, this change will reduce readability, I will re-add them.

File: scripts/includes.sh (hunk 4 above)
Comment (ttionya) [reply scripts/includes.sh:123]: This means a reference to a variable, which you can see in https://stackoverflow.com/questions/2634590/using-a-variable-to-refer-to-another-variable-in-bash

File: scripts/entrypoint.sh (hunk 1 above)
Comment (mustaphazorgati) [reply scripts/entrypoint.sh]: Thank you!

File: scripts/includes.sh (hunk 2 above)
Comment (mustaphazorgati) [reply scripts/includes.sh]: Thanks!

File: scripts/includes.sh (hunk 3 above)
Comment (mustaphazorgati) [reply scripts/includes.sh]: Thanks!

File: scripts/includes.sh (hunk 4 above)
Comment (mustaphazorgati) [reply scripts/includes.sh]: Thank you for the clarification. That example you listed shows an indirect reference where the real variable name is the value of a different variable. 

As far as I can tell we don't have a second variable which holds the reference.
```
if [[ -n "$VAR" ]]; then
```
This previous example should do the exact same thing and is (by my opinion) more readable.

Comment (mustaphazorgati) [reply scripts/includes.sh]: Thank you for clarification. 

I'd love to se a `-n` in the if/elseif statement.

**Comment by mustaphazorgati** [comment]
:clap:

**Pull request merged**